File: 0136-single-number/0136-single-number.cpp

class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int zor = 0;
        for(int i=0; i<nums.size();i++){
            zor ^= nums[i];
        }
        return zor;
    }
};





/*class Solution {
public:
    int singleNumber(vector<int>& nums) {
        unordered_map <int,int> result;
        int x;
        for(int i =0 ; i<nums.size();i++){
            int value = count(nums.begin(),nums.end(),nums[i]);
            result[nums[i] ]= value;
        }
        for(const auto &val : result){
            if (val.second == 1){
                x =  val.first;
            }
        
        }
        return x;
    }
};*/

File: 0136-single-number/0136-single-number.java
class Solution {
    public int singleNumber(int[] nums) {
        HashMap<Integer,Integer> freq = new HashMap<>();
        for(int num : nums){
            freq.put(num , freq.getOrDefault(num,0) +1);
        }
        int singleNumber =0;
        for(Map.Entry<Integer,Integer> entry: freq.entrySet()){
            if(entry.getValue() ==1){
                return entry.getKey();
            }
        }
        return 0;
    }
}

/* 
for(auto pair: freq){
    if(pair.second == 1) return pair.key;
}*/